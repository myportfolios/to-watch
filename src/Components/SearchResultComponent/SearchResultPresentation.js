import React from "react";
import "./search-result.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const SearchResultPresentation = ({
  searchMovieHandler,
  searchResult,
  selectedMoviesHandler
}) => {
  return (
    <div>
      <h2 className="sub-heading align-center">Search Movies</h2>
      <p>
        Do you have a particular movie of interest in mind? That's ok. Type in
        the title of the movie and select it from the list.
      </p>
      <SearchInput searchMovieHandler={searchMovieHandler} />
      <SearchResultTable
        searchResult={searchResult}
        selectedMoviesHandler={selectedMoviesHandler}
      />
    </div>
  );
};

export default SearchResultPresentation;

export const SearchInput = ({ searchMovieHandler }) => {
  return (
    <div className="search-icon-box">
      <input
        onKeyPress={e => {
          searchMovieHandler(e);
        }}
        type="text"
        placeholder="search movie...."
        className="text-input"
      />

      <FontAwesomeIcon icon="search" className="search-icon" />
    </div>
  );
};

const SearchResultTable = ({ searchResult, selectedMoviesHandler }) => {
  const searchResultJSX =
    searchResult &&
    searchResult.map((movie, index) => {
      return (
        <>
          <tr>
            <td>{index + 1}</td>
            <td>{movie.title}</td>
            <td>{movie.overview}</td>
            <td>{movie.release_date}</td>
            <td>{movie.vote_average}</td>
            <input
              type="checkbox"
              onClick={selectedMoviesHandler.bind(this, movie.id)}
            />
          </tr>
        </>
      );
    });
  return (
    <div>
      <table className="table">
        <thead>
          <tr>
            <th scope="col">#</th>
            <th scope="col">Title</th>
            <th scope="col">Synopsis</th>
            <th scope="col">Year</th>
            {/* <th scope="col">Actor</th> */}
            <th scope="col">Rating</th>
          </tr>
        </thead>
        <tbody>{searchResultJSX}</tbody>
        {searchResult.length > 0 && <button>Add Selected to watch list</button>}
      </table>
    </div>
  );
};
