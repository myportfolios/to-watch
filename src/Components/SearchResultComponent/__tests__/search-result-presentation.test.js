import React from "react";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import SearchResultPresentation from "../SearchResultPresentation";
import { SearchResultTable, SearchInput } from "../SearchResultPresentation";

configure({ adapter: new Adapter() });

describe("SuggestedMoviesPresentation", () => {
  let props = {
    searchMovieHandler: jest.fn(),
    searchResult: [
      {
        title: "Crucifixio"
      }
    ],
    selectedMoviesHandler: jest.fn(),

    moviesCollectionHandler: jest.fn(),
    key: ""
  };
  let wrapper = shallow(<SearchResultPresentation {...props} />);
  let wrapperSearchInput = shallow(<SearchInput {...props} />);
  let wrapperSearchResultTable = shallow(<SearchResultTable {...props} />);

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapperSearchInput).toMatchSnapshot();
    expect(wrapperSearchResultTable).toMatchSnapshot();
  });

  it("it triggers searchMovieHandler when `enter` is pressed", () => {
    wrapper.setProps({
      key: "enter"
    });
    wrapperSearchInput
      .find(".text-input")
      .simulate("keydown", { key: "enter" });
    expect(props.searchMovieHandler).toHaveBeenCalledWith({ key: "enter" });
  });

  xit("test esle statement and returns null if searchResult is empty", () => {
    wrapperSearchResultTable.setProps({
      searchResult: []
    });
    expect(wrapperSearchResultTable).toEqual({});
  });

  xit("it triggers searchMovieHandler when  `enter` is pressed", () => {
    const event = {
      target: {
        value: "abc"
      }
    };

    wrapperSearchInput
      .find(".text-input")
      .simulate("keydown", { target: { value: "abc" } });

    expect(props.searchMovieHandler).toHaveBeenCalledWith(event);
  });
});
