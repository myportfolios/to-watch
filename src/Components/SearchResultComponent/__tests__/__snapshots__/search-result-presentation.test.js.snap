// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestedMoviesPresentation renders correctly 1`] = `ShallowWrapper {}`;

exports[`SuggestedMoviesPresentation renders correctly 2`] = `ShallowWrapper {}`;

exports[`SuggestedMoviesPresentation renders correctly 3`] = `ShallowWrapper {}`;
